protocol-impl/src/main/java/io/zeebe/protocol/impl/record/CopiedRecord.java

```diff
@@ -52,6 +53,37 @@ public CopiedRecord(
     this.valueType = metadata.getValueType();
   }
 
+  private CopiedRecord(CopiedRecord<T> copiedRecord) {
+    final UnifiedRecordValue value = copiedRecord.getValue();
+    final byte[] bytes = new byte[value.getLength()];
+    final UnsafeBuffer buffer = new UnsafeBuffer(bytes);
+    value.write(buffer, 0);
+
+    final Class<? extends UnifiedRecordValue> recordValueClass = value.getClass();
+    try {
+      final T recordValue = (T) recordValueClass.newInstance();
+      recordValue.wrap(buffer);
+      this.recordValue = recordValue;
+    } catch (Exception e) {
+      throw new RuntimeException(
+          String.format(
+              "Expected to instantiate %s, but has no default ctor.", recordValueClass.getName()),
```

Comment:
Can this be the only reason for an exception? Maybe, you should make it more general.

```suggestion
              "Failed to instantiate %s.", recordValueClass.getName()),
```